packages/ag-charts-community/src/chart/series/cartesian/lineUtil.ts

```diff
@@ -223,9 +228,14 @@ export function pairCategoryData(
 
 export function determinePathStatus(newData: LineContextLike, oldData: LineContextLike) {
     let status: NodeUpdateState = 'updated';
-    if (oldData.nodeData.length === 0 && newData.nodeData.length > 0) {
+
+    const visible = (data: LineContextLike) => {
+        return data.visible;
+    };
+
+    if (!visible(oldData) && visible(newData)) {
         status = 'added';
-    } else if (oldData.nodeData.length > 0 && newData.nodeData.length === 0) {
+    } else if (visible(oldData) && !visible(newData)) {
```

Comment:
Presumable `visible` is set to `false` on a series when its `nodeData` is empty?

Reply:
So previously we were using `nodeData.length` as a way to determine visibility status, we're now being a bit more direct by driving directly from the `Series.visible` boolean flag 🤞